src/App.tsx

```diff
@@ -1,24 +1,314 @@
+                ×
+              </button>
+
+              <div
+                data-cy="TodoLoader"
+                className={classNames(
+                  'modal',
+                  'overlay',
+                  { 'is-active': todoLoader === todo.id },
+                )}
+              >
+                <div className="modal-background has-background-white-ter" />
+                <div className="loader" />
+              </div>
+            </div>
+          ))}
+
+          {tempTodo && (
+            <div
+              data-cy="Todo"
+              className="todo"
+              key={tempTodo.id}
+            >
+
+              <label className="todo__status-label">
+                <input
+                  data-cy="TodoStatus"
+                  type="checkbox"
+                  className="todo__status"
+                  checked={tempTodo.completed}
+                />
+              </label>
+
+              <span
+                data-cy="TodoTitle"
+                className="todo__title"
+              >
+                {tempTodo.title}
+              </span>
+
+              <button
+                type="button"
+                className="todo__remove"
+                data-cy="TodoDelete"
+              >
+                ×
+              </button>
+
+              <div
+                data-cy="TodoLoader"
+                className={classNames(
+                  'modal',
+                  'overlay',
+                  { 'is-active': todoLoader === 0 },
+                )}
+              >
+                <div className="modal-background has-background-white-ter" />
+                <div className="loader" />
+              </div>
+            </div>
+          )}
+        </section>
+
+        {todos.length > 0 && (
+          <footer className="todoapp__footer" data-cy="Footer">
+            <span className="todo-count" data-cy="TodosCounter">
+              {`${todos.filter(todo => !todo.completed).length} items left`}
+            </span>
+
+            <nav className="filter" data-cy="Filter">
+              <a
+                href="#/"
+                className={classNames(
+                  'filter__link',
+                  { selected: filterBy === FilterBy.All },
+                )}
+                data-cy="FilterLinkAll"
+                onClick={() => setFilterBy(FilterBy.All)}
+              >
+                All
+              </a>
+
+              <a
+                href="#/active"
+                className={classNames(
+                  'filter__link',
+                  { selected: filterBy === FilterBy.Active },
+                )}
+                data-cy="FilterLinkActive"
+                onClick={() => setFilterBy(FilterBy.Active)}
+              >
+                Active
+              </a>
+
+              <a
+                href="#/completed"
+                className={classNames(
+                  'filter__link',
+                  { selected: filterBy === FilterBy.Completed },
```

Comment:
Consider will be better if your create a helper function for this check and use it everywhere